### PR TITLE
Fix compatibility issues with MCM

### PR DESCRIPTION
**Pull request by mc2372**
The dynamically-linked Docker image may break compatibility with the MCM due to version differences.

- Statically-linked external dependencies
- Disabled unused features (e.g. font rendering)
- Switched image to `debian:bookworm-slim` to match MCM's Ruby-based image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,13 +14,23 @@ The library can be loaded from SQLite as such:
 
 ### Docker
 
-A Docker image has been provided to simplify integrating the library into larger projects. When run, an interactive SQLite session is started with the library pre-loaded for testing.
+A Docker image has been provided to simplify integrating the library into larger projects.
```
Comment (stulacy) [inline README.md]: "A Docker image has been provided containing the library statically linked against its dependencies, to simplify integrating the library into larger projects. An additional image is provided as an example of its usage"

File: README.md [hunk 2]
```diff
@@ -14,13 +14,23 @@ The library can be loaded from SQLite as such:
...
 
-A Docker image with the library pre-built is available on the GitHub Container Registry (soon) to speed up building and simplify integrating into larger projects. If you prefer, you can also build the Docker image yourself:
+- `docker/Dockerfile` builds a minimal image containing just the library. It cannot be run.
+- `docker/demo/Dockerfile` builds the library and starts an interactive SQLite demo session with it loaded.
+
+The library is available as a pre-built Docker image on the GitHub Container Registry (soon) to speed up building and simplify integrating into larger projects. If you prefer, you can also build the Docker image yourself:
```
Comment (stulacy) [inline README.md]: "Both images are provided as packages on the GitHub Container Registry (`/url/library/image` and `/url/demo/image`). If you prefer, you can also build them yourself:"

File: README.md [hunk 3]
```diff
@@ -14,13 +14,23 @@ The library can be loaded from SQLite as such:
...
 
 Run the following command:
```
Comment (stulacy) [inline README.md]: "Run the following command to build the library image:"

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+RUN cmake -DBoost_INCLUDE_DIR=/opt/boost/include -DRDK_BUILD_FREETYPE_SUPPORT=OFF -DRDK_BUILD_CFFI_LIB=ON -DRDK_BUILD_INCHI_SUPPORT=ON -DRDK_BUILD_PYTHON_WRAPPERS=OFF ..
+RUN make -j 2 cffi_test
+RUN Code/MinimalLib/cffi_test
+RUN mkdir -p $ROOT/lib && cp lib/* $ROOT/lib
+
+WORKDIR $ROOT
+RUN sed -i '/^LDFLAGS=/ s|$| -I/opt/boost/include|' Makefile
+RUN make
+RUN cp $ROOT/lib/* /lib
+RUN cp $ROOT/build/* /lib
```
Comment (stulacy) [inline docker/Dockerfile]: Where has the FROM SCRATCH mult-stage build gone? This will build a 1.9GB image right?

**Review by stulacy — COMMENTED** [review]
Thanks! Have left some comments, mostly just some wording changes, but also I can't see where the multi-stage build has gone?

File: docker/Dockerfile (hunk 4 above)
Comment (mc2372) [reply docker/Dockerfile]: Yep, you're right. I accidentally removed that line when I was moving the library-copying code to the demo

File: README.md [hunk 5]
```diff
@@ -14,13 +14,23 @@ The library can be loaded from SQLite as such:
...
+- `docker/lib/Dockerfile` builds a minimal image containing just the library. It cannot be run.
+- `docker/demo/Dockerfile` builds the library and starts an interactive SQLite demo session with it loaded.
 
-Run the following command:
+Both images are provided as packages on the GitHub Container Registry (`ghcr.io/uoy-research/rdkit-sqlite:main` and `ghcr.io/uoy-research/rdkit-sqlite-demo:main`). If you prefer, you can also build them yourself.
 
-`$ docker build -t 'rdkitsqlite' .`
+Run the following command to build the library image:
+
+`$ docker build -t 'rdkitsqlite' -f docker/lib/Dockerfile .`
```
Comment (stulacy) [inline README.md]: Can you change the tag to `rdkit-sqlite` to be consistent with how it's called in the repo

File: README.md [hunk 6]
```diff
@@ -14,13 +14,23 @@ The library can be loaded from SQLite as such:
...
+
+If you would like to run the demo, you can then build and execute it as such:
+
+```
+$ docker build -t 'rdkitsqlite-demo' -f docker/demo/Dockerfile .
```
Comment (stulacy) [inline README.md]: Ditto `rdkit-sqlite-demo`

File: README.md [hunk 7]
```diff
@@ -14,13 +14,23 @@ The library can be loaded from SQLite as such:
...
+$ docker run -it rdkitsqlite-demo
```
Comment (stulacy) [inline README.md]: Ditto `rdkit-sqlite-demo`

File: docker/demo/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,4 @@
+FROM debian:bookworm-slim
+COPY --from=rdkitsqlite /lib/librdkit* /lib/
```
Comment (stulacy) [inline docker/demo/Dockerfile]: Should this refer to the remote image we host on GHCR? I think so rather than requiring everyone to build it locally

**Review by stulacy — COMMENTED** [review]
Nice fix on the context issue. Just some consistency in naming convention comments

File: docker/demo/Dockerfile (hunk 8 above)
Comment (mc2372) [reply docker/demo/Dockerfile]: Hmm, possibly. I was coming at it from the angle of having the full build process be self-contained, but referring to the hosted image makes more sense for a "demo".

**Review by stulacy — APPROVED** [review]
Great, looks good to me!

**Pull request merged**